src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -100,19 +127,68 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 while (userName.Length == 0);
             }
 
-            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName
-            );
+            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName);
 
-            //
             // now, prompt for the password
-            //
-            WriteToConsole(passwordPrompt, true);
-            password = ReadLineAsSecureString();
-            if (password == null)
+            do
             {
-                return null;
+                WriteToConsole(passwordPrompt, true);
+                password = ReadLineAsSecureString();
+                if (password == null)
+                {
+                    return null;
+                }
             }
+            while (password.Length == 0);
+
+            if (reenterPassword)
+            {
+                reenterPasswordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_ReenterPassword, userName);
+                passwordMismatch = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_PasswordMismatch);
+
+                IntPtr pwd_ptr = IntPtr.Zero;
+                IntPtr confirmPwd_ptr = IntPtr.Zero;
+                bool isPasswordMatch = false;
+                try
+                {
+                    pwd_ptr = Marshal.SecureStringToBSTR(password);
+                    string pwd = Marshal.PtrToStringBSTR(pwd_ptr);
 
+                    while (!isPasswordMatch)
+                    {
+
+                        // now, prompt to re-enter the password.
+                        WriteToConsole(reenterPasswordPrompt, true);
+                        confirmPassword = ReadLineAsSecureString();
+                        if (confirmPassword == null)
+                        {
+                            return null;
+                        }
+
+                        confirmPwd_ptr = Marshal.SecureStringToBSTR(confirmPassword);
+                        string confirmPwd = Marshal.PtrToStringBSTR(confirmPwd_ptr);
+                        isPasswordMatch = pwd.Equals(confirmPwd);
+
+                        // If the string is not equal print message and loop.
+                        if(!isPasswordMatch)
+                        {
+                            WriteToConsole(passwordMismatch, true);
```

Comment:
Maybe use a "red" color?

Reply:
@iSazonov  I have added red color for "Passwords do not match." Since it was an error message I moved it to a separate line. Please review and let me know your feedback.
```
Get-Credential -UserName hello -ReenterPassword

PowerShell credential request
Enter your credentials.
Password for user hello: **
Re-Enter Password for user hello:
Passwords do not match.
Re-Enter Password for user hello:
Passwords do not match.
Re-Enter Password for user hello:**
Passwords do not match.
Re-Enter Password for user hello:**